### ISSUE TITLE
New Resource: EventGrid Subscriptions

### ISSUE THREAD

**Issue by gregjhogan** [terraform-providers/terraform-provider-azurerm#1794]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

There appears to currently be no support for creating azure event grid subscriptions (e.g. subscribing to storage account blob create/delete events)

### New or Affected Resource(s)

* azurerm_eventgrid_subscription (new)

### Potential Terraform Configuration

```hcl
resource "azurerm_eventgrid_subscription" "test" {
    name = "test"
    resource_group_name = "${azurerm_resource_group.test.name}"
    location = "${azurerm_resource_group.test.location}"
    topic = "${azurerm_storage_account.test.id}" # resource id
    destination = {
        endpointType = "WebHook"
        endpointUrl = "https://<webhook-url>"
        # or the following for event hub subscriptions
        #endpointType = "EventHub"
        #resourceId = ""
    }
    # filter should be optional
    filter = {
        includedEventTypes = [ "all" ] # default
        subjectBeginsWith = "" # default
        subjectEndsWith = "" # default
        isSubjectCaseSensitive = true # applies to subject begins/ends with
    }
    # labels should be optional
    labels = [ "label1", "label2" ]
}
```

### References

* https://docs.microsoft.com/en-us/rest/api/eventgrid/eventsubscriptions/createorupdate
* https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-event-quickstart?toc=%2fazure%2fevent-grid%2ftoc.json#subscribe-to-your-storage-account

**Comment by LaurentLesle** [comment]
I need it and link it to route Azure activity logs to an enventhub sitting in another subscription. 

In the EventSubscription details I want to be able to specify the Event Schema to "Same as Input Schema"

**Comment by zgqallen** [comment]
I need the similar resource, but with the end point is queue.

**Comment by timmyreilly** [comment]
As a workaround I created the Event Grid Subscription using the 'azurerm_template_deployment' resource. My use case was from Azure Storage to Event Hub. I've generalized it and uploaded a gist here: https://gist.github.com/timmyreilly/6839d0e6014167ecf68f720cb0f7794f

**Comment by phosphre** [comment]
@timmyreilly an interesting sojourn, but for other readers finding themselves here, we are wise to note the warning on the HashiCorp doc on [azurerm_template_deployment](https://www.terraform.io/docs/providers/azurerm/r/template_deployment.html) resources.

**Comment by mbfrahry** [comment]
Hey all, thanks for opening this issue. Event subscriptions for EventGrid has been merged via #2967 and will make it into the next release.
